### DISCUSSION TITLE
Hide column from editation

### DISCUSSION THREAD

**Discussion by rickball** [KevinVandy/material-react-table#849] [Q&A]
Hi, 

my second question is regarding hiding column from modal editor. Is there any way how to easily hide column from modal editor?

In case I have column with computed values, for example:

```
 {
        accessorFn: (row) => `${row.id} ${row.uuid}`,
        ...
}
```

I believe there is no reason to display it in modal crud editor. I'm able to make it read only but not easily hide it.

I found in docs mention about `columnDefType: "display",` but this means that content of the column isn't displayed in table too. So only solution is to override also Cell: {} property and re-display it again:

```
      {
        accessorFn: (row) => `${row.id} ${row.uuid}`,
        id: "uuid2",
        header: "Uuid2",
        columnDefType: "display", //turns off data column features like sorting, filtering, etc.
        Cell: ({ row, cell }) => `${cell.renderValue()}`,
      },
```

I believe it would be much easier to have some new propetry to mark column as not visibile in editation mode (for example something like `columnVisibility`) but only for editor.

Thank you

Edit PS: I now find a dirty way to filter it via `renderEditRowDialogContent`, but I need to touch inside ReactNode.:

```
   internalEditComponents = internalEditComponents.filter(c=>{
        return (c as any).props.cell.column.columnDef.accessorKey != "uuid";
      });
```

I believe there has to be easier way to do that :-)

**Comment by KevinVandy** [comment]
`Edit: () => null` in a column def

**Reply by rickball** [reply]
Thanks!

**Reply by zorbill** [reply]
Can this be done conditionally on editing or creating?  e.g. show the input on create, but not on edit?   What do I return from Edit: if I want the normal input item?
